src/Build/BuildCheck/Infrastructure/CheckWrapper.cs

```diff
@@ -52,10 +74,27 @@ internal void StartNewProject(
         }
     }
 
+    internal void ReportResult(BuildCheckResult result, ICheckContext checkContext, CheckConfigurationEffective config)
+    {
+        if (_limitReportsNumber)
+        {
+            if (_reportsCount >= MaxReportsNumberPerRule)
+            {
+                IsThrottled = true;
+                return;
+            }
```

Comment:
Let's move this check after the reporting (whit count decreased by 1) - so that we do not have an unneeded report being created and then thrown away.
After the last report we already know the check will be dismounted - we do not have to wait till the other to do that

Reply:
Thanks for fixing!